File: Algorithms/Implementation Challenges/Day of the Programmer/DayoftheProgrammer.py
#!/bin/python3

import sys

def isLeapJulian(year):
    return year % 4 == 0

def isLeapGregorian(year):
    return year % 400 == 0 or (year % 4 == 0 and year % 100 != 0)

def solve(year):
    goal = 256
    months = [31, 28, 31, 30, 31, 30, 31, 31, 30, 31]
    
    if (year < 1918 and isLeapJulian(year)) or (year > 1918 and isLeapGregorian(year)):
        months[1] = 29
    if year == 1918:
        months[1] -= 13
    month = 0
    for i in range(len(months)):
        if goal - months[i] > 0:            
            goal -= months[i]
        else:
            month = i + 1
            break
    
    return ('0' + str(goal) if goal < 10 else str(goal)) + '.' + ('0' + str(month) if month < 10 else str(month)) + '.' + str(year)

year = int(input().strip())
print(solve(year))

File: Algorithms/Implementation Challenges/Day of the Programmer/DayoftheProgrammer.java
import java.io.*;
import java.util.*;
import java.text.*;
import java.math.*;
import java.util.regex.*;

public class Solution {

    static boolean isLeapJulian(int year) {
        return year % 4 == 0;
    }
    
    static boolean isLeapGregorian(int year) {
        return year % 400 == 0 || year % 4 == 0 && year % 100 != 0;
    }

    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);
        int year = in.nextInt();
        int feb = year < 1918 && isLeapJulian(year) || year > 1918 && isLeapGregorian(year) ? 29 : year == 1918 ? 15 : 28;
        System.out.printf("%d.09.%d", 256 - 215 /* num of days in 7 months */ - feb, year);
        in.close();
    }
}
